src/main/java/mate/academy/spring/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,30 @@
+package mate.academy.spring.dao.impl;
+
+import java.util.Optional;
+import mate.academy.spring.dao.AbstractDao;
+import mate.academy.spring.dao.RoleDao;
+import mate.academy.spring.exception.DataProcessingException;
+import mate.academy.spring.model.Role;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl extends AbstractDao<Role> implements RoleDao {
+
+    public RoleDaoImpl(SessionFactory factory, Class<Role> clazz) {
+        super(factory, clazz);
+    }
+
+    @Override
+    public Optional<Role> getByName(Role.RoleName roleName) {
+        try (Session session = factory.openSession()) {
+            return Optional.ofNullable(session.createQuery("FROM Role r"
+                    + " WHERE roleName = :roleName", Role.class)
+                    .setParameter("roleName", roleName)
+                    .getSingleResult());

```

Comment:
You can use `uniqueResultOptional()` instead of `Optional.ofNullable` and `.getSingleResult()`